oak/server/rust/oak_runtime/src/node/wasm/mod.rs

```diff
@@ -824,15 +835,26 @@ impl WasmNode {
 
 /// Computes the [`NodePrivilege`] granted to a WebAssembly Node running the specified WebAssembly
 /// module.
-fn wasm_node_privilege(wasm_module_bytes: &[u8]) -> NodePrivilege {
+fn wasm_node_privilege(
+    wasm_module_bytes: &[u8],
+    wasm_module_public_key: Option<Vec<u8>>,
+) -> NodePrivilege {
     let mut hasher = sha2::Sha256::new();
-    hasher.update(&wasm_module_bytes);
-    let wasm_module_hash = hasher.finalize();
-    debug!("Wasm module SHA-256 hash: {:x}", wasm_module_hash);
-    NodePrivilege::new(
-        hashset! { oak_abi::label::web_assembly_module_tag(&wasm_module_hash) },
-        hashset! { oak_abi::label::web_assembly_module_tag(&wasm_module_hash) },
-    )
+    hasher.input(&wasm_module_bytes);
+    let module_hash = hasher.result();
+    debug!("Wasm module SHA-256 hash: {:x}", module_hash);
+    let hash_tag = oak_abi::label::web_assembly_module_tag(&module_hash);
+
+    match wasm_module_public_key {
+        Some(public_key) => {
+            let signature_tag = oak_abi::label::web_assembly_module_signature_tag(&public_key);
+            NodePrivilege::new(
+                hashset! { hash_tag.clone(), signature_tag.clone() },
+                hashset! { hash_tag, signature_tag },
+            )
+        }
+        None => NodePrivilege::new(hashset! { hash_tag.clone() }, hashset! { hash_tag }),
```

Comment:
This does not seem very elegant, one branch is a copy of the other plus more stuff. Perhaps create two separate sets, and then join then together at the end? It would also make the multiple signatures case easier to handle.

Reply:
Done